Sources/SwiftDocC/Catalog Processing/GeneratedCurationWriter.swift

```diff
@@ -0,0 +1,184 @@
+              !automaticTopics.isEmpty
+        else {
+            return nil
+        }
+        
+        let relativeLinks = linkResolver.disambiguatedRelativeLinksForDescendants(of: reference)
+        
+        // Any reference that is already curated should be excluded from the auto-generated curation
+        let alreadyCurated = Set((symbol.topics?.taskGroups ?? []).flatMap { $0.links.compactMap(\.destination) })
+        
+        // Top-level curation has a few special behaviors regarding symbols with different representations in multiple languages.
+        let isForTopLevelCuration = symbol.kind.identifier == .module
+        
+        var text = ""
+        for taskGroup in automaticTopics {
+            if isForTopLevelCuration, let firstReference = taskGroup.references.first, context.documentationCache[firstReference]?.symbol?.kind.identifier == .typeProperty {
+                // Skip type properties in top-level curation. It's not clear what's the right place for these symbols are since they exist in
+                // different places in different source languages (which documentation extensions don't yet have a way of representing).
+                continue
+            }
+            
+            let links: [(link: String, comment: String?)] = taskGroup.references.compactMap { (curatedReference: ResolvedTopicReference) -> (String, String?)? in
+                guard !alreadyCurated.contains(curatedReference.absoluteString) else { return nil }
+                guard let linkInfo = relativeLinks[curatedReference] else { return nil }
+                // If this link contains disambiguation, include a comment with the full symbol declaration to make it easier to know which symbol it refers to.
+                if linkInfo.hasDisambiguation,
+                   let symbol = context.documentationCache[curatedReference]?.symbol,
+                   let fragments = symbol[mixin: SymbolGraph.Symbol.DeclarationFragments.self]?.declarationFragments
+                {
+                    let commentText = fragments.map(\.spelling).joined().split(whereSeparator: { $0.isWhitespace || $0.isNewline }).joined(separator: " ")
+                    return (linkInfo.link, commentText)
+                }
+                return (linkInfo.link, nil)
+                
+            }
+            
+            guard !links.isEmpty else { continue }
+            
+            text.append("\n\n### \(taskGroup.title ?? "<!-- This auto-generated topic has no title -->")\n")
+            
+            let longestLink = links.map(\.link.count).max()! /* `links` are non-empty */ + 7 /* the extra characters used to make the link "\n- ``" before and "``" after */
+            for (link, comment) in links {
+                if let comment = comment {
+                    text.append("\n- ``\(link)``".padding(toLength: longestLink, withPad: " ", startingAt: 0))
```

Comment:
Is there something I am missing here, but why would whitespace padding here be useful?